### PR TITLE
Upgrade Airtable package

### PR DESCRIPTION
**Pull request by anniero98**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

Bump `airtable-schema-generator` to version 1.3.3; make necessary changes.

### REQUIRES `.env.development`/`production` to be changed from `AIRTABLE_API_KEY` to `REACT_APP_AIRTABLE_API_KEY`
...for compatibility for v1.3.3

Update Airtable schema etc - 
- **Transactions**: Rename `Total Price` --> `Total Sale`
- **Clerks**: Delete `Store ID` column (lookup); changed implementation to avoid hacky solution
- **Clerks**: Rename `Password` --> `Pin`

Removed `common.js` since we're no longer using the shared `BASE` variable for Airtable calls.

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Related PRs

Branches off of #28 
Analogous to https://github.com/calblueprint/peoplepower-web/pull/115

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review

Since changing the Airtable columns _for real_ would break PROD, you have to change the Clerks table columns temporarily to test Clerk login; and change the Transactions table temporarily to test checkout flow.

We'll probably need to change them all at once after this is merged and the build is finished / App store approves 😅 The solution to this is likely to have a PROD Airtable base - you might want to look into that @wangannie 

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps

N/A

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

See "How to review"

### Screenshots

N/A

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,2 +1,8 @@
 AIRTABLE_BASE_ID='app4fXK49bqcjDMEo'
-AIRTABLE_API_KEY='your-api-key-here'
\ No newline at end of file
+
+## Named to conform to airtable-schema-generator's expectations
```
Comment (aivantg) [inline .env.example:3]: \>:o

File: .eslintrc.js [hunk 2]
```diff
@@ -1,23 +1,46 @@
...
     ecmaVersion: 2018,
-    sourceType: 'module'
+    sourceType: 'module',
   },
   plugins: ['react-native', 'prettier'],
   rules: {
+    'no-restricted-imports': [
+      'error',
+      {
+        paths: [
```
Comment (aivantg) [inline .eslintrc.js:24]: If you wanted to be even cooler (like, fang should deng cool), you could restrict request.js outside of utility files....

File: lib/airtable/airtable.js [hunk 3]
```diff
@@ -29,180 +28,166 @@ const base = Airtable.base(BASE_ID);
...
-          // No need for this to throw an error, sometimes there's just no values
-          // reject(new Error(`No record was retrieved using this ${fieldType}.`));
-          return;
-        }
-
-        resolve(records.map(record => fromAirtableFormat(record.fields, table)));
-      });
-  });
+function getRecordsByAttribute(table, fieldType, field, sort = []) {
+  return base(table)
```
Comment (aivantg) [inline lib/airtable/airtable.js:143]: Wowow so clean. Such beautiful promises

File: lib/airtable/request.js [hunk 4]
```diff
@@ -85,112 +77,147 @@ export const getCustomersById = async id => {
...
-export const getAllClerkss = async () => {
-  return getAllRecords(Tables.Clerks);
+export const getClerksByIds = async ids => {
+  const formula = `OR(${ids.reduce((f, id) => `${f} {ID}='${id}',`, '')} 1 < 0)`;
+  return getAllRecords(Tables.Clerks, formula);
 };
 
-export const getClerkssByStoreId = async value => {
-  return getRecordsByAttribute(Tables.Clerks, Columns[Tables.Clerks].storeId, value);
+export const getAllClerks = async (filterByFormula = '', sort = []) => {
```
Comment (aivantg) [inline lib/airtable/request.js:115]: Look at that beAUTIFUL pluralization

File: lib/airtable/schema.js [hunk 5]
```diff
@@ -7,164 +7,142 @@ export const Tables = {
...
   },
   'Push Tokens': {
-    primaryKey: `Primary Key`,
-    created: `Created`,
-    customerId: `Customer`,
-    customerName: `Customer Name`,
-    token: `Token`,
-    id: `id`,
+    primaryKey: { name: `Primary Key`, type: `formula` },
+    created: { name: `Created`, type: `formula` },
```
Comment (aivantg) [inline lib/airtable/schema.js:37]: Such expressive types 😍

File: lib/checkoutUtils.js [hunk 6]
```diff
@@ -52,21 +52,22 @@ export async function addTransaction(customer, cart, transaction) {
...
-    totalPrice: totalSale,
+    totalSale,
   });
 
   // A list of ids for line items from the transaction.
   const itemIds = await calculateProductsPurchased(cart);
 
-  await updateTransactions(transactionId, { productsPurchasedId: itemIds });
+  // productsPurchaseIds - airtable-schema-generator depluralizing bug
+  await updateTransactions(transactionId, { productsPurchaseIds: itemIds });
```
Comment (aivantg) [inline lib/checkoutUtils.js:70]: sad :/

File: lib/loginUtils.js [hunk 7]
```diff
@@ -1,22 +1,23 @@
...
   return stores;
 }
 
 // lookupClerk searches for clerks based on the store ID (from the picker) and numeric password.
 // If the user is found, we return the record. Otherwise, we will display an error on the login screen.
-export async function lookupClerk(storeId, password) {
-  const records = await getClerkssByStoreId(storeId);
+export async function lookupClerk(storeId, pin) {
+  const currStore = await getStoresById(storeId);
+  const clerks = await getClerksByIds(currStore.clerkIds);
```
Comment (aivantg) [inline lib/loginUtils.js:13]: This code makes me happy

File: package.json [hunk 8]
```diff
@@ -16,8 +16,16 @@
     "generate-schema": "generate-airtable-schema"
   },
   "airtable-schema-generator": {
-    "input": "lib/airtable",
-    "output": "lib/airtable"
+    "output": "lib/airtable",
+    "mode": "auto-headless",
+    "envFileName": ".env.development",
+    "schemaMeta": {
```
Comment (aivantg) [inline package.json:22]: Wow such customizable

File: lib/checkoutUtils.js (hunk 6 above)
Comment (anniero98) [reply lib/checkoutUtils.js:70]: HMM should I.... fix it?

File: lib/loginUtils.js (hunk 7 above)
Comment (anniero98) [reply lib/loginUtils.js:13]: I looked at yours 👀

File: lib/checkoutUtils.js (hunk 6 above)
Comment (aivantg) [reply lib/checkoutUtils.js:70]: maybe you could create an ISSUE
